src/admin/dumb_components/member_payments.js

```diff
@@ -47,6 +47,6 @@ var MemberPayments = (
     </div>
   </div>
 
-const with_amount = (type, defaults) => type === 'subscription' ? defaults : dissoc('amount', defaults)
+const with_amount = (type, defaults) => type === 'subscription' ? defaults : assoc('amount', '', defaults)
```

Comment:
Yeah field `amount` was being populated no matter what type of payment was selected.